tests/cpp/test_gpu3.cpp

```diff
@@ -8236,6 +8236,59 @@ TEST_F(NVFuserTest, FusionCpAsyncPredicateError) {
           ::testing::HasSubstr("unsupported use case of cp.async")));
 }
 
+TEST_F(NVFuserTest, DecoupledDomains) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // XX shape structure:
+  //
+  // domain 0: [I0, I1...    I2  I3} domain 1
+  //             \  /         \  /
+  //            merge         merge
+  //             /  \         /  \.
+  // domain 1: {I4  I5    ...I6, I7] domain 0
+  // where domain 0 is [I0, I1, I6, I7], and
+  //       domain 1 is [I4, I5, I2, I3]
+  auto create_xx_shape_structure = []() {
+    auto s0 = IrBuilder::create<Val>(DataType::Index);
+    auto s1 = IrBuilder::create<Val>(DataType::Index);
+    auto s2 = IrBuilder::create<Val>(DataType::Index);
+    auto s3 = IrBuilder::create<Val>(DataType::Index);
+    auto id0 = IterDomainBuilder(s0->fusion()->zeroVal(), s0).build();
+    auto id1 = IterDomainBuilder(s1->fusion()->zeroVal(), s1).build();
+    auto id2 = IterDomainBuilder(s2->fusion()->zeroVal(), s2).build();
+    auto id3 = IterDomainBuilder(s3->fusion()->zeroVal(), s3).build();
+    AbstractTensor dom0({id0, id1, id2, id3});
+    AbstractTensor dom1 = dom0;
+    dom0.merge(2);
+    dom0.split(2, 256);
+    dom1.merge(0);
+    dom1.split(0, 256);
+    return std::make_pair(dom0.as<IterDomain*>(), dom1.as<IterDomain*>());
+  };
+  auto [root_xx0, root_xx1] = create_xx_shape_structure();
```

Comment:
It seems to me that these domains are not tied to be root domains. They are just two domains derived from the same domain but differently. They are used as part of root, logical and allocation domains. Am I understanding it correctly? If so, I found calling them "root_xx" a little confusing.